spec/PeriodicBackgroundSync-index.bs

```diff
@@ -92,3 +94,95 @@ Fetch requests within the <code>periodicsync</code> event while [=in the backgro
+    1. Let <var>timeTillScheduledEventForOrigin</var> be the time when the next periodicSync event is scheduled for the origin, if any, null otherwise.
+    1. If |timeTillScheduledEventForOrigin| is not null:
+        1. If |timeTillScheduledEventForOrigin| - |delayForOrigin| is greater than or equal to |minIntervalForOrigin|, abort these substeps.
+        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| and abort these substeps.
+        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|.
+    1. Let <var>minIntervalAcrossOrigins</var> be the minimum interval allowed between periodicSync events across origins. Let <var>timeSinceLastPeriodicSync</var> be the time since a periodicSync event was fired for any origin, if any, null otherwise.
+    1. If |timeSinceLastPeriodicSync| is null, set |timeToFire| to |delayForOrigin| + |now|.
+    1. Else:
+        1. If |timeSinceLastPeriodicSync| is greater than equal to |minIntervalAcrossOrigins|, set |timeToFire| to |delayForOrigin| + |now|.
+        1. Else, set <var>timeTillNextAllowedPeriodicSync</var> to |minIntervalAcrossOrigins| - |timeSinceLastPeriodicSync|. Set |timeToFire| to the maximum of |delayForOrigin| + |now|, and |timeTillNextAllowedPeriodicSync| + |now|.
+    1. Set the [=time to fire=] of |registration| to |timeToFire|.
+1. Else:
+    1. Let <var>countRetries</var> be the count of retries that have been made for this event.
+    1. Increment |countRetries|.
+    1. If |countRetries| is equal to the [=maximum number of retries=] allowed, set |countRetries| to 0 and follow the steps to [=first attempt of firing|fire the first attempt=].
+    1. Else, Set the [=time to fire=] of |registration| to |now| + a small back-off that is exponentially proportional to |countRetries|.
+
+API Description {#api-description}
+===================================
+## Extensions to the {{ServiceWorkerGlobalScope}} interface ## {#extensions-to-serviceworkerglobalscope}
+<script type="idl">
+partial interface ServiceWorkerGlobalScope {
+    attribute EventHandler onperiodicsync;
+};
+</script>
+
+## Extensions to the {{ServiceWorkerRegistration}} interface ## {#extensions-to-serviceworkerregistration}
+
+<script type="idl">
+[Exposed=(Window,Worker)]
+partial interface ServiceWorkerRegistration {
+  readonly attribute PeriodicSyncManager periodicSync;
+};
+</script>
+
+<div dfn-for="ServiceWorkerRegistration">
+A {{ServiceWorkerRegistration}} has a <dfn>periodic sync manager</dfn> (a {{PeriodicSyncManager}}), initially a new {{PeriodicSyncManager}} whose [=service worker registration=] is the [=context object=]'s [=service worker registration=].
```

Comment:
Is this [=service worker registration=] pointing to the one you defined? I don't think so since it isn't scoped.

Reply:
where have a I defined one? I'd actually avoid redefining it if I can reuse it from the serviceworkers spec.

Reply:
In that case you should have a definition somewhere that a PeriodicSyncManager has an associated variable of type [=service worker registration=], and you set the to the context object's ...

Reply:
I think that by virtue of adding a PeriodicSyncManager to a ServiceWorkerRegistration, a 1:1 relationship has already been established between the two. Therefore, I can unambiguously refer to the [=service worker registration=] of the PeriodicSyncManager, without needing to store that value in a variable.
For a similar example, see the background-fetch spec:
https://wicg.github.io/background-fetch/#extensions-to-service-worker-registration

Reply:
Not quite, it's defined here: https://wicg.github.io/background-fetch/#backgroundfetchmanager-service-worker-registration. And when you click on the link it takes you to the variable declaration rather than to the Service Worker specification

Reply:
ah, fair. I'll define one too then.

Reply:
done. I've also updated references in the rest of the spec.